assets/wizards/engagement/views/newsletters/index.js

```diff
@@ -1,91 +1,225 @@
 /**
- * Collects and connects with Mailchimp API key.
+ * Internal dependencies
  */
+import { values, startCase, uniq, mapValues, property } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useState, useEffect, Fragment } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
 import { __ } from '@wordpress/i18n';
 import { ExternalLink } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import {
-	Card,
 	Notice,
 	TextControl,
+	CheckboxControl,
+	SelectControl,
 	PluginInstaller,
 	withWizardScreen,
+	SectionHeader,
+	Button,
+	Grid,
+	hooks,
 } from '../../../../components/src';
+import { fetchJetpackMailchimpStatus } from '../../../../utils';
 
-/**
- * Initial connection to Mailchimp.
- */
-class Newsletters extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			pluginRequirementsMet: false,
-		};
-	}
-	/**
-	 * Render.
-	 */
-	render() {
-		const { apiKey, connected, connectURL, wcConnected, onChange } = this.props;
-		const { pluginRequirementsMet } = this.state;
-		return [
-			<Card key="mailchimp-block">
-				{ ! connected && (
-					<p>
-						{ __(
-							'This feature connects your site to Mailchimp and sets up a Mailchimp block you can use to get new subscribers for your newsletter. The Mailchimp connection to your site for this feature is managed through Jetpack and WordPress.com.'
-						) }
-					</p>
+export const NewspackNewsletters = ( { className, onUpdate, mailchimpOnly = true } ) => {
+	const [ config, updateConfig ] = hooks.useObjectState( {} );
+	const [ allProviders, setAllProviders ] = useState( [] );
+	const performConfigUpdate = update => {
+		updateConfig( update );
+		if ( onUpdate ) {
+			onUpdate( mapValues( update.settings, property( 'value' ) ) );
+		}
+	};
+	useEffect( () => {
+		apiFetch( {
+			path: '/newspack/v1/wizard/newspack-engagement-wizard/newsletters',
+		} ).then( res => {
+			setAllProviders(
+				uniq( res.settings.reduce( ( acc, setting ) => [ ...acc, setting.provider ], [] ) ).filter(
```

Comment:
When I visit the Engagement > Newsletters wizard, I only see Mailchimp as an option despite the `newsletters` endpoint returning the correct options.

<img width="1843" alt="Screen Shot 2021-02-23 at 1 23 54 PM" src="https://user-images.githubusercontent.com/2230142/108903541-a49cec80-75da-11eb-91e6-848825bd6c26.png">